chess/state-transition/src/stf/v1/persist/lobby.ts

```diff
@@ -11,15 +11,37 @@ import { persistNewRound } from './match.js';
 import type { SQLUpdate } from '@paima/node-sdk/db';
 import { initialState } from '@chess/game-logic';
 import { schedulePracticeMove } from './practice.js';
+import { Events, CreatedLobby_v1, CreatedLobby_v2 } from '@chess/events';
+import { encodeEventForStf } from '@paima/events';
+import { precompiles } from '@chess/precompiles';
 
 // Persist creation of a lobby
 export function persistLobbyCreation(
   player: WalletAddress,
   blockHeight: number,
   inputData: CreatedLobbyInput,
-  randomnessGenerator: Prando
+  randomnessGenerator: Prando,
+  events: Events
```

Comment:
I think we shouldn't return data by modifying the reference, it works - but as there is no way to tell the programmer this will happen (as in other languages that you can mark as mutable) in the worst case perhaps call it `mutableEvents`

Reply:
That's just how I prefer to write it though, there is nothing enforcing one style or the other. I personally always assume that if I pass an array then it may be mutated (in js). In this case there is not really any point in reading from this object, so it has to be write only, but that can be clearer through the name as you mention.

Reply:
This is just a personal preference thing and it's up to each game to decide how they want to do this. This PR is more to decide if the interface of the things coming from Paima Engine itself is good or not